### PR TITLE
Fixing issues needed to support Scala 2.13

### PR DESCRIPTION
**Pull request by amir734jj**
Description:
1) HashSet is a final class now so we can't extend it
  - `findEntry` method doesn't exist, we need to use `find` instead

2) ~~- Extending a Set class requires implementing the `diff` method~~
- In scala 2.13 it is required to implement the `diff` method so we have to add the `override` keyword
- In scala 2.12 if we have an `override` keyword we get "error: method diff overrides nothing"
  
   We can either use a macro to add the `override` keyword depending on scala version or just remove a `RangeSet` which is not used anywhere.

3) '+' operator between char and string has been deprecated. Use `toString()` instead
4) procedure syntax has been deprecated: `def foo() { }` // no equal sign
5) Typo in the comment

Build status:

```
scalac 2.9: couldn't get scala 2.9 to install locally (getting link error with JVM)
scalac 2.10: built successfully
scalac 2.11: built successfully
scalac 2.12: built successfully
scalac 2.13: built successfully
```

### CODE REVIEW

File: src/edu/uwm/cs/util/RangeSet.scala [hunk 1]
```diff
@@ -17,6 +17,7 @@ class RangeSet(val range : Range) extends Set[Int]
   override def -(elem : Int) : Set[Int] = throw new UnsupportedOperationException("RangeSet#-");
   override def +(elem : Int) : Set[Int] = throw new UnsupportedOperationException("RangeSet#+");
   override def iterator : Iterator[Int] = range.iterator;
+  override def diff(that: Set[Int]): Set[Int] = this.diff(that)
```
Comment (boyland) [inline src/edu/uwm/cs/util/RangeSet.scala]: Isn't this an infinite recursion ?

File: src/edu/uwm/cs/util/Reach.scala [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/* Dtermining which nodes are in cycles in a graph.
+/* Determining which nodes are in cycles in a graph.
```
Comment (boyland) [inline src/edu/uwm/cs/util/Reach.scala]: The comment before and after seems to be wrong.  It was copied from Cycle.scala, but this file doesn't do cycle detection, does it?

File: src/edu/uwm/cs/util/MyHashSet.scala [hunk 3]
```diff
@@ -1,8 +0,0 @@
-package edu.uwm.cs.util
-
-import scala.collection.mutable.HashSet
-
-class MyHashSet[T] extends HashSet[T] {
```
Comment (boyland) [inline src/edu/uwm/cs/util/MyHashSet.scala:5]: Getting rid of this hand-override is a good idea.  I'm happy that HashSet provides what you need and you found it.

**Review by boyland — COMMENTED** [review]
These changes, needed for 2.13, seem like they should work for earlier versions too.

**Comment by amir734jj** [comment]
Build status: 
- scalac 2.9: couldn't get scala 2.9 to install locally
- scalac 2.10: built successfully
- scalac 2.11:  built successfully
- scalac 2.12:  built successfully
- scalac 2.13:  built successfully

**Review by boyland — APPROVED** [review]
Looks good.

**Pull request merged**